packages/app/pages/account/password/[token].tsx

```diff
@@ -0,0 +1,198 @@
+import React, { ReactElement, useEffect, useState } from 'react'
+import Head from 'next/head'
+import {
+  Button,
+  Card,
+  Form,
+  Input,
+  Spin,
+  message,
+  Typography,
+  Result,
+} from 'antd'
+import { StandardPageContainer } from '../../../components/common/PageContainer'
+import { useRouter } from 'next/router'
+import styled from 'styled-components'
+
+export default function ForgetPasswordReset(): ReactElement {
+  const router = useRouter()
+  const { token } = router.query
+  const [isTokenValid, setIsTokenValid] = useState(null)
+  const [invalidTokenMessage, setInvalidTokenMessage] = useState(null)
+  const [resetPasswordForm] = Form.useForm()
+
+  const Container = styled.div`
+    margin-left: auto;
+    margin-right: auto;
+    text-align: center;
+    padding-top: 100px;
+    width: 60%;
+    border-radius: 15px;
+    height: auto;
+
+    @media (max-width: 650px) {
+      width: 90%;
+    }
+
+    @media (max-width: 992px) {
+      width: 80%;
+    }
+  `
+
+  useEffect(() => {
+    fetch(`/api/v1/auth/password/reset/validate/${token}`).then(
+      async (response) => {
+        if (response.ok) {
+          setIsTokenValid(true)
+        } else {
+          const data = await response.json()
+
+          setInvalidTokenMessage(data.message)
+        }
+      },
+    )
+  })
+
+  function ResponseContainer(): ReactElement {
+    const resetPassword = async () => {
+      const formValues = await resetPasswordForm.validateFields()
+      const { passwordField, confirmPasswordField } = formValues
+
+      if (passwordField !== confirmPasswordField) {
+        message.error('Passwords do not match')
+        return
+      }
+
+      const request = {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify({
+          password: passwordField,
+          confirmPassword: confirmPasswordField,
+        }),
+      }
+
+      fetch(`/api/v1/auth/password/reset/${token}`, request).then(
```

Comment:
why did you do the requests like this instead of `API.auth.functionName()`?

Reply:
I found this to be more reasonable to use to fetch error and properly display what went wrong. Axios library did not return exact error message that server has returned.